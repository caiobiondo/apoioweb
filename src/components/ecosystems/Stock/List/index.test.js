import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import StockList from './index';

const baseState = {
  empty: false,
  loading: true,
  productSearch: '',
};

describe('StockList Ecosystem', () => {
  it('should render the stock list page', () => {
    const renderer = new ShallowRenderer();

    renderer.render(<StockList />);
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });

  it('should update product search filter on state', () => {
    const productSearch = 'a name';
    const expectedState = { ...baseState, productSearch };

    const result = shallow(<StockList />);
    const instance = result.instance();
    instance.onSearch({ name: productSearch });

    expect(instance.state).toEqual(expectedState);
  });

  it('should update empty and loading state', () => {
    const expectedState = { ...baseState, empty: true, loading: false };

    const result = shallow(<StockList />);
    const instance = result.instance();
    instance.onLoadFinished(true, false);

    expect(instance.state).toEqual(expectedState);
  });
});
