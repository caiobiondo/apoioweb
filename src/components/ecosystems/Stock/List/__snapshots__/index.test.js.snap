// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockList Ecosystem should render the stock list page 1`] = `
<indexstyles__Main>
  <Apollo(StockAddProductModal)
    handleClose={[Function]}
    opened={false}
    user={undefined}
  />
  <Apollo(ListTable)
    onLoadFinished={[Function]}
    productSearch=""
  />
</indexstyles__Main>
`;
